Gem/Code/Source/Automation/ScriptReporter.cpp

```diff
@@ -685,6 +577,154 @@ namespace AtomSampleViewer
+                            ScriptReporter::FindBestToleranceLevel(screenshotResult.m_officialComparisonResult.m_standardDiffScore, false);
+
+                        ImGui::Text(
+                            "(Unfiltered Diff Score: %f%s)", screenshotResult.m_officialComparisonResult.m_standardDiffScore,
+                            unfilteredTolerance ? AZStd::string::format(" ~ '%s'", unfilteredTolerance->m_name.c_str()).c_str() : "");
+                    }
+                }
+
+                ResetTextHighlight();
+
+                ImGui::PushID("Official");
+                ShowDiffButton("View Diff", screenshotResult.m_officialBaselineScreenshotFilePath, screenshotResult.m_screenshotFilePath);
+                ImGui::PopID();
+
+                if ((m_forceShowExportPngDiffButtons ||
+                     screenshotResult.m_officialComparisonResult.m_resultCode == ImageComparisonResult::ResultCode::ThresholdExceeded) &&
+                    ImGui::Button("Export Png Diff"))
+                {
+                    const AZStd::string imageDiffPath = GenerateAndCreateExportedImageDiffPath(scriptReport, screenshotResult);
+                    ExportImageDiff(imageDiffPath.c_str(), screenshotResult);
+                    m_messageBox.OpenPopupMessage(
+                        "Image Diff Exported Successfully",
+                        AZStd::string::format("The image diff file was saved in %s", imageDiffPath.c_str()).c_str());
+                }
+
+                if ((!screenshotPassed || m_forceShowUpdateButtons) && ImGui::Button("Update##Official"))
+                {
+                    if (screenshotResult.m_localComparisonResult.m_resultCode == ImageComparisonResult::ResultCode::FileNotFound)
+                    {
+                        UpdateSourceBaselineImage(screenshotResult, true);
+                    }
+                    else
+                    {
+                        m_messageBox.OpenPopupConfirmation(
+                            "Update Official Baseline Image",
+                            "This will replace the official baseline image \n"
+                            "with the image captured during this test run. \n"
+                            "Are you sure?",
+                            // It's important to bind screenshotResult by reference because UpdateOfficialBaselineImage will update it
+                            [this, &screenshotResult]()
+                            {
+                                UpdateSourceBaselineImage(screenshotResult, true);
+                            });
+                    }
+                }
+            }
+            ImGui::Unindent();
+
+            ImGui::Spacing();
+
+            HighlightTextIf(localBaselineWarning, m_highlightSettings.m_highlightWarning);
+
+            ImGui::Text("Local Baseline:    %s", screenshotResult.m_localBaselineScreenshotFilePath.c_str());
+
+            // Local Baseline Result
+            ImGui::Indent();
```

Comment:
Would recommend extracting out functions into something like ShowLocalBaselineResult() for instance.